node/src/services/blockchain/handler.rs

```diff
@@ -272,12 +274,21 @@ impl BlockchainService {
         rpc_handlers: Arc<RpcHandlers>,
         keystore: KeystorePtr,
     ) -> Self {
+        // Get best nonce (not pending) for one of the pub keys (in this case the last one)
+        // in our vector of keys in the Keystore.
+        let current_block_hash = client.chain_info().best_hash;
```

Comment:
Although better than before, I believe this won't solve our problems definitively. This code executes before the node syncs the chain with its peers. Meaning that if there was a transaction for this `pub_key` after this node's last block recorded (which when initialising from scratch is 0), this won't return the correct nonce.

Reply:
This is a good point. I changed it so we start with 0 and update the nonce after each new block notification is identified by the service.